src/common/function/src/scalars/json/json_path_exists.rs

```diff
@@ -64,32 +78,63 @@ impl Function for JsonPathExistsFunction {
         let paths = &columns[1];
 
         let size = jsons.len();
-        let datatype = jsons.data_type();
         let mut results = BooleanVectorBuilder::with_capacity(size);
 
-        match datatype {
-            // JSON data type uses binary vector
-            ConcreteDataType::Binary(_) => {
+        match (jsons.data_type(), paths.data_type()) {
+            (ConcreteDataType::Binary(_), ConcreteDataType::String(_)) => {
                 for i in 0..size {
                     let json = jsons.get_ref(i);
                     let path = paths.get_ref(i);
 
-                    let json = json.as_binary();
-                    let path = path.as_string();
-                    let result = match (json, path) {
-                        (Ok(Some(json)), Ok(Some(path))) => {
-                            let json_path = jsonb::jsonpath::parse_json_path(path.as_bytes());
-                            match json_path {
-                                Ok(json_path) => jsonb::path_exists(json, json_path).ok(),
-                                Err(_) => None,
+                    // Get json as bytes.
+                    let json = match json.as_binary() {
+                        Ok(Some(bytes)) => bytes,
+                        _ => {
+                            return InvalidFuncArgsSnafu {
+                                err_msg: format!("Illegal json binary: {:?}", json),
                             }
+                            .fail();
                         }
-                        _ => None,
+                    };
+
+                    let result = if !path.is_null() {
+                        // Get path as str.
+                        let path_str = match path.as_string() {
+                            Ok(Some(str)) => str,
+                            _ => {
+                                return InvalidFuncArgsSnafu {
+                                    err_msg: format!("Unable to parse path as string: {:?}", path),
+                                }
+                                .fail();
+                            }
```

Comment:
I don't think we need to handle this solution. The path must be a string.

Reply:
cc @CookiePieWw

Reply:
We can simply put a `None` here as other udf does.
https://github.com/GreptimeTeam/greptimedb/blob/191755fc42f45ecc3628c55427e7e5abbecd2ede/src/common/function/src/scalars/date/date_format.rs#L94-L103

Reply:
Yes, I prefer the original logic.